primitives/utils/src/metrics.rs

```diff
@@ -0,0 +1,147 @@
+
+use lazy_static::lazy_static;
+use prometheus::{
+    Registry, Error as PrometheusError, Opts,
+    core::{
+        GenericGaugeVec, AtomicU64,
+        GenericGauge,
+	}
+};
+use std::time::Instant;
+use std::convert::TryFrom;
+use parking_lot::Mutex;
+
+use crate::channels::{ Sender, Receiver, OnFullStrategy, channel_with_strategy };
+
+lazy_static! {
+
+    pub static ref TOKIO_THREADS_TOTAL: GenericGauge<AtomicU64> = GenericGauge::new(
```

Comment:
```suggestion
    pub static ref TOKIO_THREADS_TOTAL: GenericCounter<AtomicU64> = GeneicCounter::new(
```

If I understand correctly this is a monotonic increasing counter.